### PR TITLE
Updating node version chore

### PR DESCRIPTION
**Pull request by gkwang**
Update to node 12 with a major version bump. 
Updated deprecated Buffer use.

### CODE REVIEW

File: admin/server.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ app.use(cookieParser());
 app.use(session({
   secret: 'sojo sut ed oterces le'
 }));
+app.use(test)
```
Comment (madsharm) [inline admin/server.js:34]: What is the purpose of this?

File: package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "connect-multiparty": "^2.2.0",
     "cookie-parser": "^1.4.3",
     "cookie-sessions": "github:auth0/cookie-sessions#53a8aae",
+    "diff_match_patch": "^0.1.1",
```
Comment (madsharm) [inline package.json:35]: Are we using it anywhere?

Comment (gkwang) [reply package.json:35]: Yes, it's used but was undeclared, that's why I added it.

File: admin/server.js (hunk 1 above)
Comment (gkwang) [reply admin/server.js:34]: weird I don't remember adding it. It's not suppose to be there. Let me fix it.

File: package.json (hunk 2 above)
Comment (gkwang) [reply package.json:35]: Used here: https://github.com/auth0/ad-ldap-connector/blob/0d3fda67dcfafd60de9feeced1d994913dffdd67/admin/public/codemirror-addon/merge/merge.js#L8

**Review by madsharm — APPROVED** [review]

File: installer/prepare.ps1 [hunk 3]
```diff
@@ -21,7 +21,7 @@ $msiDependencies = @(
     } ;
     @{
         'Name'='nodejs';
-        'Url'="https://nodejs.org/dist/v4.4.7/node-v4.4.7-x86.msi";
+        'Url'="https://nodejs.org/dist/v12.19.0/node-v12.19.0-x86.msi";
```
Comment (siacomuzzi) [inline installer/prepare.ps1:24]: did you test the connector with node v12?

Comment (gkwang) [reply installer/prepare.ps1:24]: yes, it works on node 12. However I still need to test out the installer, but I can't do so without jenkins building it.

**Review by siacomuzzi — APPROVED** [review]

**Review by siacomuzzi — APPROVED** [review]

**Comment by siacomuzzi** [comment]
image

please remember to update Jenkins configuration too.

**Comment by siacomuzzi** [comment]
There are more required changes to include in this PR, see https://github.com/auth0/ad-ldap-connector/pull/175/commits/1999a95429c42cf51ba18601e63fd9c0f9196036 (please let me know if you have some questions)

**Review by siacomuzzi — CHANGES_REQUESTED** [review]
There are more required changes to include in this PR, see https://github.com/auth0/ad-ldap-connector/pull/175/commits/1999a95429c42cf51ba18601e63fd9c0f9196036 (please let me know if you have some questions)

**Comment by siacomuzzi** [comment]
@gkwang is there an ETA for this task? We have another PR to fix vulnerabilities reported by npm and snyk, but we have to merge it after this PR.

**Comment by siacomuzzi** [comment]
could you please run the tests locally (`npm test`)? Skipping https://github.com/auth0/ad-ldap-connector/blob/master/test/users.tests.js which needs a config.json file in place pointing to an external test-AD.

**Comment by gkwang** [comment]
@siacomuzzi   tests passed after removing ./lib/tls and the associated test.

File: test/certs.js [hunk 4]
```diff
@@ -32,28 +31,10 @@ describe('Connection to server with empty subject', function () {
     }
   });
 
-  // This test is mainly to keep track of the original Node.Js issue. When this test fails you can safely remove ../lib/tls.js
```
Comment (siacomuzzi) [inline test/certs.js:35]: > When this test fails you can safely remove ../lib/tls.js
could you please remove all references to `./lib/tls.js` and SSL_ENABLE_EMPTY_SUBJECT?

File: admin/test_config.js [hunk 5]
```diff
@@ -111,11 +110,6 @@ module.exports = function (config, callback) {
       tlsOptions = {
         ca: cas
       };
-
-      if (nconf.get('SSL_ENABLE_EMPTY_SUBJECT')) {
```
Comment (siacomuzzi) [inline admin/test_config.js:115]: please remove all references to SSL_ENABLE_EMPTY_SUBJECT and `lib/tls`

File: test/certs.js (hunk 4 above)
Comment (gkwang) [reply test/certs.js:35]: done.

File: admin/test_config.js (hunk 5 above)
Comment (gkwang) [reply admin/test_config.js:115]: done.

File: lib/ldap.js [hunk 6]
```diff
@@ -14,10 +14,6 @@ function createConnection () {
 
   if (nconf.get('LDAP_URL').toLowerCase().substr(0, 5) === 'ldaps') {
     clientOpts.tlsOptions = { ca: https.globalAgent.options.ca };
-    if (nconf.get('SSL_ENABLE_EMPTY_SUBJECT')) {
```
Comment (siacomuzzi) [inline lib/ldap.js:17]: please remove line 8 (`./tls`), it was removed, right?

File: troubleshoot.js [hunk 7]
```diff
@@ -266,7 +266,7 @@ async.series([
 			host,
 			port: port || 636,
 			ca: https.globalAgent.options.ca,
-			checkServerIdentity: nconf.get('SSL_ENABLE_EMPTY_SUBJECT') && tls.checkServerIdentity
+			checkServerIdentity: tls.checkServerIdentity
```
Comment (siacomuzzi) [inline troubleshoot.js:269]: I guess you can remove this line, right?

**Comment by siacomuzzi** [comment]
please update installer/README.md  too

**Comment by siacomuzzi** [comment]
LGTM

was you able to generate and test the msi installer?

**Comment by gkwang** [comment]
@siacomuzzi  it doesn't look like I have permission to change the build branch in jenkins. I pushed this PR change to `node12-upgrade` if you can build it, I can test it out.

**Review by siacomuzzi — APPROVED** [review]

**Pull request merged**